File: Laboratórios/Java/ex1_slide_herancaMutlipla/src/ex1_slide_herancaMutlipla/Livro.java
package ex1_slide_herancaMutlipla;

public class Livro extends Material {

	protected int numero_de_paginas;

	public Livro(String titutlo, String autor, int numPags) {
		super(titutlo, autor);
		this.numero_de_paginas = numPags;
	}

	public int getNumero_de_paginas() {
		return numero_de_paginas;
	}

	public void setNumero_de_paginas(int numero_de_paginas) {
		this.numero_de_paginas = numero_de_paginas;
	}

	@Override
	public void descricao() {
		super.descricao();
		System.out.printf("Número de páginas: %d\n", this.numero_de_paginas);
	}
	
	
	
}

File: Testes/ex1_slide_herancaMultipla/Livro.cpp
#include "Livro.hpp"

Livro::Livro(string titulo, string autor, int numPags)
    :Material(titulo, autor), numero_de_paginas(numPags){}

void Livro::descricao()
{
    Material::descricao();
    cout<< "Número de páginas: " << numero_de_paginas << endl;
}